stories/LinkComponent-stories.tsx

```diff
@@ -0,0 +1,100 @@
+/*
+ * Echoes React
+ * Copyright (C) 2023-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+import { Link } from '@sonarsource/echoes-react';
+import type { Meta, StoryObj } from '@storybook/react';
+import { OpenNewTabIcon } from '../src/common/icons/OpenNewTabIcon';
+
+const meta: Meta<typeof Link> = {
+  component: Link,
+  title: 'Links',
+  args: {
+    to: '/path',
+  },
+} satisfies Meta<typeof Link>;
+
+export default meta;
+
+type Story = StoryObj<typeof meta>;
+
+export const LinkDefault: Story = {
+  args: {
+    children: 'Link Default',
+    to: '/path',
+  },
+};
+
+export const LinkDefaultWithIcon: Story = {
+  args: {
+    children: 'Link Default with Icon',
+    to: '/path',
+    icon: <OpenNewTabIcon />,
+  },
+};
+
+export const LinkExternal: Story = {
+  args: {
+    children: 'Link External',
+    to: 'https://abc.com/path/new',
+  },
+};
+
+export const LinkInsideParagraphWithStyle: Story = {
+  args: {
+    children: 'Link External inside paragraph',
+    to: 'https://abc.com/path/new',
+  },
+  render: (args) => (
+    <p style={{ color: 'grey', fontSize: '20px', fontWeight: '100' }}>
+      <Link {...args} />
+    </p>
+  ),
```

Comment:
What I meant was surround it with text, like:
```
<p>
Hello, click <Link /> to go to the good place.
</p>
```

But your example already reveals an issue with the Icon alignment:
![image](https://github.com/SonarSource/echoes-react/assets/47277647/0b9eba5a-2045-4e54-a9ae-8fcfb61e0bf9)
(or maybe it's the font size that triggers that?)

Reply:
I think we shouldn't care about that for now.
Our products still only support single-size icon of 16px only.

We should anyway revisit a bit the implementation of this when we implement the new icons with the 3 sizes and the typography components that should be used for sizing things.

Maybe just reduce the font-size back to 1rem on this story for now and use a different font-decoration instead like italic.

Reply:
By the way, I created DS-202 for that.